component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -108,7 +113,14 @@ public class TOTPUtil {
      */
     public static String encrypt(String plainText) throws CryptoException {
 
-        return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(plainText.getBytes(Charsets.UTF_8));
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        // Get custom key from server configuration.
+        String customKey = null;
+        if (SUPER_TENANT_DOMAIN.equals(tenantDomain)) {
```

Comment:
Are we doing this custom encryption for super tenant users only?

Reply:
Yes. this is only for asgardeo users since we are sharing them across region.